src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -475,6 +458,22 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool c
             }
         }
 
+        if (getPetType() == SUMMON_PET && !current) //all (?) summon pets come with full health when called, but not when they are current
+        {
+            SetPower(POWER_MANA, GetMaxPower(POWER_MANA));
+            SetFullHealth();
+        }
+        else
+        {
+            if (!savedhealth && getPetType() == HUNTER_PET)
+                setDeathState(JUST_DIED);
```

Comment:
Should check if pet isn't dead, could be possible causing new issue

Reply:
Okay, thanks